Library/Homebrew/formula.rb

```diff
@@ -2429,10 +2430,45 @@ def to_hash_with_variations
       end
     end
 
-    hash["variations"] = variations
+    hash["variations"] = @variations_hash
     hash
   end
 
+  # Builds internal json v3 hash.
+  #
+  # @private
+  def to_api_hash
+    api_hash = to_hash_with_variations.except(
+      # Included in the top-level of the `homebrew-core.json` file
+      # created by the `brew generate-formula-api` command.
+      "aliases",
+      "oldname",
+      "oldnames",
+      "tap",
+      "tap_git_head",
+    )
+
+    # Remove keys that default to zero.
```

Comment:
Could maybe also do this for keys that are `null` (if you haven't already) e.g. `urls.stable.using` for most formulae

Reply:
There were diminishing returns with trying to remove nested blank fields in the previous PR but I can just double-check that right now.